test_conformance/common/vulkan_wrapper/opencl_vulkan_wrapper.cpp

```diff
@@ -846,7 +867,83 @@ clExternalSemaphore::~clExternalSemaphore() noexcept(false)
     }
 }
 
-int clExternalSemaphore::signal(cl_command_queue cmd_queue)
+int clExternalImportableSemaphore::wait(cl_command_queue cmd_queue)
```

Comment:
We can now wait on ImportableSemaphore, but need to be able to signal too for other handle types.
Can we extend these to support signal operations?

Reply:
The choice to not implement both is not related to handle types. The unimplemented methods are never used by any test.

Reply:
Unimplemented methods are never used now because clVk2CLExternalSemaphore is made clExternalImportableSemaphore object and clCl2VkExternalSemaphore is made clExternalExportableSemaphore object compared to earlier when both were just plain clExternalSemaphore objects where both wait and signal would have been used. If exporting semaphores is not supported on some platform, there is no way to run the tests in the current form compared to earlier when they could have run with just imported semaphores.

Reply:
Is there a concrete example, or more of a future thing?  I can implement these method, but unless we add a test they will not get used.  Implementing them was not a spec choice, just a choice not to write unused code.  

It is functionally the same as the old system, except it reflects the fact that semaphores cannot be both importable and exportable.  This was necessary, the other option would have been to add a new parameter to clExternalSemaphore() called import_export_mode or something, otherwise the constructor would have no idea what to do with the vulkan semaphore being passed in.